File: go/0226/0226_InvertBinaryTree.go
// 226. Invert Binary Tree

package main

import (
	"bufio"
	"leetcode/structs"
	"leetcode/utils"
	"os"

	"github.com/samber/lo"
)

func invertTree(root *structs.TreeNode) *structs.TreeNode {
	if root != nil {
		root.Left, root.Right = root.Right, root.Left
		invertTree(root.Left)
		invertTree(root.Right)
	}
	return root
}

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	var lines []string
	for scanner.Scan() {
		lines = append(lines, scanner.Text())
	}

	const testSize int = 1

	for _, test := range lo.Chunk(lines, testSize) {
		root := utils.ToTree(test[0])
		result := invertTree(root)
		utils.PrintTreeInOrder(result)
	}
}

File: py/0226_InvertBinaryTree.py
# 226. Invert Binary Tree

from utils import chunk
from utils import chunk, toTree, printTree
from typing import Optional
from classes import TreeNode


class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if not root:
            return
        root.left, root.right = root.right, root.left
        self.invertTree(root.left)
        self.invertTree(root.right)
        return root


if __name__ == "__main__":
    testSize = 1
    puzzles = [
        [4, 2, 7, 1, 3, 6, 9],
        [2, 1, 3],
        [],
    ]
    for puzzle in chunk(puzzles, testSize):
        printTree(Solution().invertTree(toTree(puzzle[0])))
        print()
